From 575f48b5b6c8fc070dde274c4b9f6bf7b5c2abd6 Mon Sep 17 00:00:00 2001
From: Scott Graham <sgraham@gmail.com>
Date: Thu, 9 Jan 2025 15:28:55 -0800
Subject: [PATCH 02/10] fix -Wsometimes-uninitialized warning in non DEBUG
 builds

---
 gcm.c       | 2 +-
 rv64/isel.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcm.c b/gcm.c
index e685d2a..cef1ddf 100644
--- a/gcm.c
+++ b/gcm.c
@@ -144,7 +144,7 @@ schedlate(Fn *fn, Ref r)
 	Use *u;
 	uint earlybid;
 	uint latebid;
-	uint uselatebid;
+	uint uselatebid = 0;
 
 	if (rtype(r) != RTmp)
 		return NOBID;
diff --git a/rv64/isel.c b/rv64/isel.c
index ca62359..dace4c4 100644
--- a/rv64/isel.c
+++ b/rv64/isel.c
@@ -155,7 +155,7 @@ selcmp(Ins i, int k, int op, Fn *fn)
 		i.op = KWIDE(k) ? Oceqd : Oceqs;
 		break;
 	default:
-		assert(0 && "unknown comparison");
+		die("unknown comparison");
 	}
 	if (op < NCmpI)
 		i.op = sign ? Ocsltl : Ocultl;
-- 
2.50.1 (Apple Git-155)

