From 0f74b8601ce3e0fcec7ca17bf5bd225314398ba2 Mon Sep 17 00:00:00 2001
From: Scott Graham <scott.github@h4ck3r.net>
Date: Tue, 14 Jan 2025 16:50:50 -0800
Subject: [PATCH 03/18] Add printtyp() and debug['T']

---
 all.h   |  1 +
 main.c  |  1 +
 parse.c | 40 ++++++++++++++++++++++++++++++++++++++++
 3 files changed, 42 insertions(+)

diff --git a/all.h b/all.h
index 9e1e633..961b4fe 100644
--- a/all.h
+++ b/all.h
@@ -539,6 +539,7 @@ bshas(BSet *bs, uint elt)
 extern Op optab[NOp];
 void parse(FILE *, char *, void (char *), void (Dat *), void (Fn *));
 void printfn(Fn *, FILE *);
+void printtyp(Typ*, FILE *);
 void printref(Ref, Fn *, FILE *);
 void err(char *, ...) __attribute__((noreturn));
 
diff --git a/main.c b/main.c
index ed1ac94..ebde93c 100644
--- a/main.c
+++ b/main.c
@@ -17,6 +17,7 @@ char debug['Z'+1] = {
 	['L'] = 0, /* liveness */
 	['S'] = 0, /* spilling */
 	['R'] = 0, /* reg. allocation */
+	['T'] = 0, /* types */
 };
 
 extern Target T_amd64_sysv;
diff --git a/parse.c b/parse.c
index 406d13f..25a0e1a 100644
--- a/parse.c
+++ b/parse.c
@@ -1071,6 +1071,11 @@ parsetyp(void)
 	} else
 		parsefields(ty->fields[n++], ty, t);
 	ty->nunion = n;
+
+  if (debug['T']) {
+    fprintf(stderr, "\n> Parsed type:\n");
+    printtyp(ty, stderr);
+  }
 }
 
 static void
@@ -1269,6 +1274,41 @@ printcon(Con *c, FILE *f)
 	}
 }
 
+static char* field_type_name[] = {
+    [FEnd] = "End",  //
+    [Fb] = "b",      //
+    [Fh] = "h",      //
+    [Fw] = "w",      //
+    [Fl] = "l",      //
+    [Fs] = "s",      //
+    [Fd] = "d",      //
+    [FPad] = "Pad",  //
+    [FTyp] = "Typ",  //
+};
+
+void printtyp(Typ* ty, FILE* f) {
+  fprintf(f, "type :%s\n", ty->name);
+  fprintf(f, "  isdark: %d\n", ty->isdark);
+  fprintf(f, "  isunion: %d\n", ty->isunion);
+  fprintf(f, "  align: %d\n", ty->align);
+  fprintf(f, "  size: %u\n", (uint)ty->size);
+  fprintf(f, "  nunion: %u\n", ty->nunion);
+  for (uint i = 0; i < ty->nunion; ++i) {
+    fprintf(f, "  union %d\n", i);
+    Field* fld = ty->fields[i];
+    for (int j = 0; j < NField+1; ++j) {
+      if (fld[j].type == FTyp) {
+        fprintf(f, "    field %d: %s => %s\n", j, field_type_name[fld[j].type],
+                typ[fld[j].len].name);
+      } else {
+        fprintf(f, "    field %d: %s len %u\n", j, field_type_name[fld[j].type], fld[j].len);
+      }
+      if (fld[j].type == FEnd)
+        break;
+    }
+  }
+}
+
 void
 printref(Ref r, Fn *fn, FILE *f)
 {
-- 
2.53.0.vfs.0.0

