From 56064191fd64cdfdb7b8ef8ba8119c9de8280ccf Mon Sep 17 00:00:00 2001
From: Scott Graham <sgraham@gmail.com>
Date: Thu, 9 Jan 2025 15:28:55 -0800
Subject: [PATCH 03/10] fix -Wsometimes-uninitialized warning in non DEBUG
 builds

---
 rv64/isel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rv64/isel.c b/rv64/isel.c
index 000c112..0fb85b3 100644
--- a/rv64/isel.c
+++ b/rv64/isel.c
@@ -154,7 +154,7 @@ selcmp(Ins i, int k, int op, Fn *fn)
 		i.op = KWIDE(k) ? Oceqd : Oceqs;
 		break;
 	default:
-		assert(0 && "unknown comparison");
+		die("unknown comparison");
 	}
 	if (op < NCmpI)
 		i.op = sign ? Ocsltl : Ocultl;
-- 
2.49.0.vfs.0.4

