From d0818e602ba7d55ffc0f6d3f8f7d16486e9593b6 Mon Sep 17 00:00:00 2001
From: Scott Graham <scott.github@h4ck3r.net>
Date: Thu, 5 Feb 2026 14:23:20 -0800
Subject: [PATCH 13/18] avoid warnings for variables that are only used for
 asserts

---
 gvn.c  | 1 +
 rega.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/gvn.c b/gvn.c
index 8cd52b4..2881e09 100644
--- a/gvn.c
+++ b/gvn.c
@@ -219,6 +219,7 @@ assoccon(Fn *fn, Blk *b, Ins *i1)
 	|| (i1->cls == Kw && (int32_t)c.bits.i < 0)) {
 		fail = negcon(i1->cls, &c);
 		assert(fail == 0);
+		(void)fail;
 		op = Osub;
 	}
 
diff --git a/rega.c b/rega.c
index 48d6637..279e207 100644
--- a/rega.c
+++ b/rega.c
@@ -257,6 +257,7 @@ move(int r, Ref to, RMap *m)
 	if (bshas(m->b, r)) {
 		/* r is used and not by to */
 		assert(r1 != r);
+		(void)r1;
 		for (n=0; m->r[n] != r; n++)
 			assert(n+1 < m->n);
 		t = m->t[n];
-- 
2.53.0.vfs.0.0

